File: Programmers/Python/60057.py
def solution(s):
    result = len(s)
    for c in range(1, len(s)//2+1) :
        result = min(length(s, c), result)
    return result

def length(s, c):
    start = 0
    length = len(s)
    accCnt = 1
    for start in range(c, len(s), c) :
        if s[start-c:start] == s[start:start+c] :
            accCnt += 1
        else :
            if accCnt > 1 :length -= (c*(accCnt-1) - len(str(accCnt)))
            accCnt = 1
    return length if accCnt == 1 else length - (c*(accCnt-1) - len(str(accCnt)))
        

File: Programmers/Swift/60057.swift
import Foundation

func solution(_ s:String) -> Int {
    return Array(1...max(1, Int(s.count/2))).reduce(s.count) { b, c in min(b, checkCount(s, c)) }
}

func checkCount(_ s: String, _ c: Int) -> Int {
    let string = Array(s)
    var length = s.count
    var start = 0
    var accumulateCount = 1
    while start+2*c <= s.count {
        var valid = true
        for i in start..<start+c where string[i] != string[i+c] { 
            valid = false
            break
        }
        switch valid {
        case true : accumulateCount += 1
        case false : 
            length -= (accumulateCount > 1 ? c*(accumulateCount-1) - (String(accumulateCount).count) : 0)
            accumulateCount = 1
        }
        start += c
    }
    length -= (accumulateCount > 1 ? c*(accumulateCount-1)-String(accumulateCount).count : 0)
    return length
}